kin-devplatform-sdk/src/main/java/kin/devplatform/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -225,12 +223,25 @@ public void run() {
 		});
 	}
 
+	@Override
+	public void reconnectBalanceConnection() {
+		synchronized (balanceObserversLock) {
```

Comment:
Why do we need a synchronized block here?
Do we have a concern about many thread that can execute this code?
Is it a preparation for multiple account or something like that?

Reply:
that's simply multiple threads supporting, no relation what so ever to multiple account support...